### PR TITLE
apidoc error/warning fix (#6)

### PR DESCRIPTION
**Pull request by ebarboni**
This is a sixth split of the big PR https://github.com/apache/netbeans/pull/4450.
using the following to have ant javadoc -Dmetabuild.branch=master -Dapidocfullcheck=true.

About 3/4 of the module pass.

I did a comment on a code modification that is removal of import. I don't know why it fails with this import. Many classes have unused import and no issue.

As part of the refactoring.api fixes it seems that annotationprocessor is supporting a never implemented type. (see inline comment)

### CODE REVIEW

File: ide/refactoring.api/src/org/netbeans/modules/refactoring/spi/ui/ScopeProvider.java [hunk 1]
```diff
@@ -72,7 +71,6 @@ public abstract class ScopeProvider {
      *
      * @see ScopeProvider#initialize(org.openide.util.Lookup,
      * java.util.concurrent.atomic.AtomicBoolean)
-     * @see ScopeDescription
      * @return an icon for the Scope, or null if the default should be used.
```
Comment (ebarboni) [inline ide/refactoring.api/src/org/netbeans/modules/refactoring/spi/ui/ScopeProvider.java:74]: this class was never implemented. 
was intended first https://bz.apache.org/netbeans/show_bug.cgi?id=217347#a0 (see attachement).

https://github.com/apache/netbeans/blob/master/ide/refactoring.api/src/org/netbeans/modules/refactoring/spi/impl/ScopeAnnotationProcessor.java could be fixed to remove the SupportedAnnotationTypes ScopeDescription

File: java/java.source.base/src/org/netbeans/modules/java/source/JavadocHelper.java [hunk 2]
```diff
@@ -70,7 +70,6 @@
 import javax.swing.text.html.HTML;
 import javax.swing.text.html.HTMLEditorKit;
 import javax.swing.text.html.parser.ParserDelegator;
-import org.netbeans.api.*;
 import org.netbeans.api.annotations.common.CheckForNull;
```
[the remark this answers is not in the document]
Comment (mbien) [reply java/java.source.base/src/org/netbeans/modules/java/source/JavadocHelper.java:73]: always good to avoid star imports

**Review by mbien — APPROVED** [review]
looked through it and everything looks good to me.

please wait with merge till master is green again.

File: java/java.source.base/src/org/netbeans/modules/java/source/JavadocHelper.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (BradWalker) [reply java/java.source.base/src/org/netbeans/modules/java/source/JavadocHelper.java:73]: firmly agree!

**Comment by ebarboni** [comment]
master is green merging

**Pull request merged**